vnext/Microsoft.ReactNative/ReactNativeHost.cpp

```diff
@@ -54,7 +48,11 @@ void ReactNativeHost::InstanceSettings(ReactNative::ReactInstanceSettings const
   m_instanceSettings = value;
 }
 
-void ReactNativeHost::ReloadInstance() noexcept {
+IAsyncAction ReactNativeHost::LoadInstance() noexcept {
+  return ReloadInstance();
+}
+
+IAsyncAction ReactNativeHost::ReloadInstance() noexcept {
```

Comment:
To me this seems a bit counter-intuitive...
To me it seems backwards. `LoadInstance` and `UnloadInstance` would have implemenataton. and `ReloadInstance` would just call `UnloadInstance` and then `LoadInstance`.

I realize it is an artifact of the m_reactHost api... just calling out if feels 'funny'.

Reply:
The ReloadInstance is much more powerful than LoadInstance because otherwise requiring to always unload before load would be too difficult and error prone. I would rather have it as the prime API instead of inventing what to do if developers call LoadInstance twice without calling UnloadInstance between them. Should we crash or ignore? Plus, the Reload call could let us optimize some internals if needed.
So, why to have LoadInstance in addition to the ReloadInstance which does the same work? The answer is the "syntactic sugar". The ReloadInstance call looks strange when we just start application. Saying LoadInstance seems to be more logical. Internally the 'unload' part is noop in that case anyway.

---
In reply to: [432967474](https://github.com/microsoft/react-native-windows/pull/5071#discussion_r432967474) [](ancestors = 432967474)